tungsten_ci_utils/nexus_cleanup/clean_up.groovy

```diff
@@ -0,0 +1,117 @@
+import org.sonatype.nexus.repository.storage.StorageFacet;
+import org.sonatype.nexus.common.app.GlobalComponentLookupHelper
+import org.sonatype.nexus.repository.maintenance.MaintenanceService
+import org.sonatype.nexus.repository.storage.ComponentMaintenance
+import org.sonatype.nexus.repository.storage.Query;
+import org.sonatype.nexus.script.plugin.RepositoryApi
+import org.sonatype.nexus.script.plugin.internal.provisioning.RepositoryApiImpl
+import com.google.common.collect.ImmutableList
+import org.joda.time.DateTime;
+import org.slf4j.Logger
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+
+def retentionCount = 15;
+def retentionDays = 30;
+def repositoryName = 'BartsDockerRepo';
+def whitelistCore = ["latest", "ocata", "newton", "queens"].toArray();
+def whitelist = ["latest","40", "94", "122", "129", "161", "214", "309", "360"].toArray();
+
+log.info(":::Cleanup script started!");
+MaintenanceService service = container.lookup("org.sonatype.nexus.repository.maintenance.MaintenanceService");
+def repo = repository.repositoryManager.get(repositoryName);
+def tx = repo.facet(StorageFacet.class).txSupplier().get();
+def components = null;
+try {
+    tx.begin();
+    components = tx.browseComponents(tx.findBucket(repo));
+}catch(Exception e){
+    log.info("Error: "+e);
+}finally{
+    if(tx!=null)
+        tx.close();
+}
+
+if(components != null) {
+    def retentionDate = DateTime.now().minusDays(retentionDays).dayOfMonth().roundFloorCopy();
+    int deletedComponentCount = 0;
+    int compCount = 0;
+    def listOfComponents = ImmutableList.copyOf(components);
+    def previousComp = listOfComponents.head().name();
+    listOfComponents.reverseEach{comp ->
+
+        log.info("Processing Component - ${comp.name()}, version: ${comp.version()}");
+        log.info("----------");
+        def fruitEnd = "null";
+        for(i = 0; i < whitelistCore.length; i++){
```

Comment:
think this loop can be substituted in the following manner

``
 def comp_version = 'rhel-queens-master-395'

 def whitelistCore = ["latest", "ocata", "newton", "queens" ].toArray(); 
 
 def fruitJuice = whitelistCore.any{ item -> item.contains(comp_version)}
 
 print fruitJuice
``